clang/lib/Sema/Sema.cpp

```diff
@@ -1715,6 +1715,11 @@ class DeferredDiagnosticsEmitter
               << Sema::KernelConstStaticVariable;
           return;
         }
+        if (!VD->hasInit() &&
+            S.isTypeDecoratedWithDeclAttribute<SYCLGlobalVariableAllowedAttr>(
```

Comment:
Shouldn't this be `SYCLDeviceGlobalAttr` . `SYCLGlobalVariableAllowedAttr` can be used to mark other types in the future.

Reply:
I think you are right. I've made the change.

Reply:
This sounds right.